### PR TITLE
[show][config] add muxcable command line support for retrieve / reset ICMP packet loss data

### PR DESCRIPTION
**Pull request by zjswhhh**
Stemming from https://github.com/Azure/sonic-linkmgrd/pull/14
sign-off: Jing Zhang zhangjing@microsoft.com

#### What I did
Added support to retrieve and reset ICMP packet loss data in state db for muxcable.  

#### How I did it
Changes made in show/muxcable.py and config/muxcable.py

#### How to verify it
- Added unit tests. 
- Tested the command lines on testbeds. 
- Ran dualtor_io/test_link_failure.py. 

#### New command output (if the output of a command-line utility has changed)
``` show muxcable pckloss <port_name>```
```
admin@str2-7050cx3-acs-07:~$ show muxcable packetloss Ethernet96 
PORT        COUNT                 VALUE
----------  ------------------  -------
Ethernet96  pck_loss_count        10439
Ethernet96  pck_expected_count    11406
PORT        EVENT                      TIME
----------  -------------------------  ---------------------------
Ethernet96  link_prober_unknown_start  2022-Jan-27 19:47:17.819699
Ethernet96  link_prober_unknown_end    2022-Jan-27 22:28:36.736928
```
```config muxcable pckloss reset <port_name>```
```
admin@str2-7050cx3-acs-07:~$ sudo config muxcable packetloss reset Ethernet96
admin@str2-7050cx3-acs-07:~$ show muxcable packetloss Ethernet96
PORT        COUNT                 VALUE
----------  ------------------  -------
Ethernet96  pck_expected_count        0
Ethernet96  pck_loss_count            0
PORT        EVENT                      TIME
----------  -------------------------  ---------------------------
Ethernet96  link_prober_unknown_start  2022-Jan-27 19:47:17.819699
Ethernet96  link_prober_unknown_end    2022-Jan-27 22:28:36.736928
```
```config muxcable pckloss reset all```
```
admin@str2-7050cx3-acs-07:~$ sudo config muxcable packetloss reset all
admin@str2-7050cx3-acs-07:~$ show muxcable packetloss Ethernet68
PORT        COUNT                 VALUE
----------  ------------------  -------
Ethernet68  pck_loss_count            0
Ethernet68  pck_expected_count        3
PORT        EVENT                      TIME
----------  -------------------------  ---------------------------
Ethernet68  link_prober_unknown_start  2022-Jan-27 19:47:17.702760
Ethernet68  link_prober_unknown_end    2022-Jan-27 22:28:36.756113

```

### CODE REVIEW

File: config/muxcable.py [hunk 1]
```diff
@@ -333,6 +341,72 @@ def mode(db, state, port, json_output):
...
+        else:
+            click.echo("there is not a valid asic table for this asic_index".format(asic_index))
+            sys.exit(CONFIG_FAIL)
+
+    elif port == "all" and port is not None:
+
+        for namespace in namespaces:
+            asic_id = multi_asic.get_asic_index_from_namespace(namespace)
+            for key in port_table_keys[asic_id]:
+                logical_port = key.split("|")[1]
```
Comment (vdahiya12) [inline config/muxcable.py:405]: Here I think we should check against the port being in "MUX_CABLE" config db table

**Review by vdahiya12 — CHANGES_REQUESTED** [review]

File: config/muxcable.py (hunk 1 above)
Comment (zjswhhh) [reply config/muxcable.py:406]: Updated accordingly.

File: config/muxcable.py [hunk 2]
```diff
@@ -333,6 +341,72 @@ def mode(db, state, port, json_output):
         sys.exit(CONFIG_SUCCESSFUL)
 
 
+ #'muxcable' command ("config muxcable pckloss reset <port|all>")
+@muxcable.command()
+@click.argument('action', metavar='<action_name>', required=True, type=click.Choice(["reset"]))
+@click.argument('port', metavar='<port_name>', required=True, default=None)
+@clicommon.pass_db
+def pckloss(db, action, port):
```
[the remark this answers is not in the document]
Comment (zjswhhh) [reply config/muxcable.py]: updated accordingly.

**Comment by zjswhhh** [comment]
@vdahiya12 - can I have your approval on this PR? 

@qiluo-msft - can we force merge this PR as it has been blocked for over a week?

**Review by vdahiya12 — APPROVED** [review]

**Comment by qiluo-msft** [comment]
This commit could not be cleanly cherry-picked to 202012. Please submit another PR.

**Pull request merged**